src/libcommonserver/CMakeLists.txt

```diff
@@ -83,9 +87,11 @@ endif()
 # Link
 target_link_libraries(${libcommonserver_NAME}
     ${libcommon_NAME}
+    ${libparms_NAME}
```

Comment:
Don't link libparms to libcommonserver.
Create a new class "LogArchiver" at "server" level with the log archiving functionalities.

Reply:
As discussed, I implemented this new code architecture in commit: https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/5797be9b81e5c345ea7b0a9d0e0512bd244287f3